Generative-AI/Youtube_Rag_Chatbot/main.py

```
@@ -0,0 +1,125 @@
+    try:
+        # Chain setup
+        parallel_chain = RunnableParallel({
+            "context": retriever | RunnableLambda(format_docs),
+            "question": RunnablePassthrough()
+        })
+
+        chain = parallel_chain | prompt | gemini | StrOutputParser()
+        result = chain.invoke(question)
+        return {"answer": result.strip()}
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Error while processing question: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)